src/commons/api/roas.rs

```diff
@@ -280,22 +286,186 @@ impl Ord for RoaDefinition {
     }
 }
 
-impl PartialOrd for RoaDefinition {
+impl PartialOrd for RoaPayload {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         Some(self.cmp(other))
     }
 }
 
-impl AsRef<TypedPrefix> for RoaDefinition {
+impl AsRef<TypedPrefix> for RoaPayload {
     fn as_ref(&self) -> &TypedPrefix {
         &self.prefix
     }
 }
 
+//------------ RoaConfiguration --------------------------------------------
+
+/// This type defines a saved RoaConfiguration. It includes the actual ROA
+/// payload that needs be authorized in a ROA object as well as other
+/// information that is only visible to users - i.e. an optional comment
+/// field, and in future perhaps other things such as tags used for
+/// classification/monitoring/bpp analysis.
+///
+/// Note that this type is backward compatible with the RoaDefinition type
+/// used until Krill 0.10.0.
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct RoaDefinitions(Vec<RoaDefinition>);
+pub struct RoaConfiguration {
+    #[serde(flatten)]
+    payload: RoaPayload,
+    #[serde(default)] // missing is same as no comment
+    comment: Option<String>,
+}
+
+impl RoaConfiguration {
+    pub fn new(payload: RoaPayload, comment: Option<String>) -> Self {
+        RoaConfiguration { payload, comment }
```

Comment:
Is it handled somewhere that we ensure that comments cannot contain an (unescaped?) bare `#` character, as that character is used to prefix the comment in the `Display` impl?

Reply:
The implementation currently uses the *first* # on the line to indicate the start of a comment section (which may be empty even btw). Subsequent # are accepted inside comments.

Reply:
I don't think this is an issue for json as it's inside a string. I think serde will take care of " characters inside the string..

We could limit the allowed characters, but I am afraid that that will severely limit the usability for languages other than English.